src/search.cpp

```diff
@@ -278,8 +278,7 @@ Query& Query::setGeorange(float latitude, float longitude, float distance) {
 
 int Search::getEstimatedMatches() const
 {
-    auto&& lock = mp_internalDb->lock();
-    (void)lock;
+    auto lock = SearchLockGuard(*mp_internalDb);
```

Comment:
Why not `SearchLockGuard lock(*mp_internalDb);`?

Reply:
I tend to use auto "everywhere" since I have seen this video : https://www.youtube.com/watch?v=xnqTKD8uD64&t=2406s

In instance, with `auto lock = SearchLockGuard(*mp_internalDb)`, you don't care what actually is `SearchLockGuard`, it my be a function or a type, we don't care. We just want to get a search lock guard.

With `SearchLockGuard lock(*mp_internalDb)`, you assert that `SearchLockGuard` is a type. If one day we refactor `SearchLockGuard` to be a function, we would have to modify all the call site.